pkg/types/aws/machinepool.go

```diff
@@ -24,6 +24,11 @@ type MachinePool struct {
 	//
 	// +optional
 	EC2RootVolume `json:"rootVolume"`
+
+	// The name of a user-created IAM Role to use. This takes the place of an installer-created role.
+	//
+	// +optional
+	IAMRoleName string `json:"iamRoleName,omitempty"`
```

Comment:
The name part seems unnecessary.
```suggestion
	IAMRole string `json:"iamRole,omitempty"`
```